src/phoenix/experimental/evals/models/openai.py

```diff
@@ -22,6 +32,48 @@
 }
 LEGACY_COMPLETION_API_MODELS = ("gpt-3.5-turbo-instruct",)
 logger = logging.getLogger(__name__)
+TQDM_BAR_FORMAT = (
+    "Eta:{eta} |{bar}| {percentage:3.1f}% "
+    "({n_fmt}/{total_fmt}) "
+    "[{elapsed}<{remaining}, {rate_fmt}{postfix}]"
+)
+
+
+def openai_token_usage(chat_completion: OpenAIObject) -> int:
+    try:
+        return int(chat_completion.usage.total_tokens)
+    except (AttributeError, ValueError):
+        return 0
+
+
+def openai_rate_limit_info(model_name: str, api_key: Optional[str]) -> Mapping[str, int]:
```

Comment:
Azure needs to be excluded from this function, because the URL, request headers, and json payload are all slightly different, and the Azure response headers don't contain the `x-ratelimit-` fields that I can see.

Reply:
Are Azure models called from this interface as well? I thought this model was specifically for the OpenAI API

Reply:
Yea, unfortunate they do use the same model. The OpenAI API refers to the LLM API, not the HTTP API. The LLM API is the same, but the HTTP is not.

Reply:
I'm adding some basic handling for this, I think we can/should do more research later on a better way